histogram/histogram.go

```diff
@@ -18,12 +18,12 @@ type Config struct {
 // See https://godoc.org/github.com/prometheus/client_golang/prometheus#MustNewConstHistogram.
 type Histogram struct {
 	// count is the number of entries added to the Histogram.
-	count uint64
+	count int32
 	// sum is the total sum of all entries added to the Histogram.
 	sum float64
 	// buckets is a map of upper bounds to cumulative counts of entries,
 	// excluding the +Inf bucket.
-	buckets map[float64]uint64
+	buckets map[float64]int32
```

Comment:
I think the map management here is the root cause and I do not see how we solve anything now. It will only take longer for the OOM to happen again. When is the map flushed? Should we re-initialize it after a week or whatever?

Reply:
The keys for this map are fixed at initialisation, so I don’t see why we’d need to flush it, it won’t grow over time.

Reply:
To be specific, I agree that this wouldn’t fin any OOM issues, it just helps reduce the overall necessary memory. More a nice to have :D

Reply:
I see, but then I am wondering how this can contribute to any OOM. How many bytes does the change here safe? I would assume that the OOM is about megabytes.

Reply:
I recall we saw those 512kB being allocated for `buckets`, @JosephSalisbury even did some quick math which explained how those would sum up to megabytes.

Reply:
Yeah, we use a fair few of these in the net-exporter, and it adds up.